test/suites/greenboot.robot

```diff
@@ -0,0 +1,123 @@
+*** Settings ***
+Documentation       Tests related to greenboot
+Resource            ../resources/common.resource
+Resource            ../resources/systemd.resource
+Resource            ../resources/ostree-health.resource
+Library    SSHLibrary
+Suite Setup         Setup Suite 
+Suite Teardown      Teardown Suite
+
+
+*** Variables ***
+${LOCAL_CONFIG_FILE}                  ../docs/config/busybox_running_check.sh
+${REMOTE_CONFIG_FILE}                 /etc/greenboot/check/required.d/50_busybox_running_check.sh
+${GREENBOOT_CONFIG_FILE}              /etc/greenboot/greenboot.conf
+${GREENBOOT_CONFIG_CONTENT}           MICROSHIFT_WAIT_TIMEOUT_SEC=180
+${MANIFEST_SOURCE_DIR}                assets/kustomizations/greenboot/
+${MANIFEST_DIR}                       /etc/microshift/manifests
+${MICROSHIFT_CONFIG}                  /etc/microshift/config.yaml
+*** Test Cases ***
+
+# user-workload-health
+
+GreenbootHealth Run with User Workload
+    [Documentation]    check
+    Upload User Config
+    Microshift Cleanup And Start
+    Cleanup User Config
+
+GreenbootHealth Microshift Service Failure
+    [Documentation]    Microshift Service failure
+    Microshift Disrupt Service
+    Run Keyword And Expect Error    	0 != 1
+    ...    Microshift Cleanup And Start
+    Microshift Restore Service
+    Microshift Cleanup And Start
+
+GreenbootHealth Microshift Pod Failure
+    [Documentation]    Microshift Pod Network failure
+    Microshift Disrupt Pod Network
+    Run Keyword And Expect Error    	0 != 1
+    ...    Greenboot Healthy
+    Microshift Cleanup And Start
+
+
+
+*** Keywords ***
+Setup Suite
+    [Documentation]    Set up all of the tests in this suite
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig
+    Update Greenboot Config
+
+Update Greenboot Config
+    [Documentation]    Upload a new configuration file to the MicroShift host
+    Upload String To File   ${GREENBOOT_CONFIG_CONTENT}    ${GREENBOOT_CONFIG_FILE}
+
+Upload User Config
+    [Documentation]    Upload a new configuration file to the MicroShift host
+    Put File    ${LOCAL_CONFIG_FILE}    /tmp/busybox_running_check.sh
+    ${stdout}    ${rc}=    Execute Command
+    ...    [ -f ${REMOTE_CONFIG_FILE} ] || sudo ln -s /tmp/busybox_running_check.sh ${REMOTE_CONFIG_FILE} 
```

Comment:
If the command is already elevated, do we need the sudo?

Reply:
since we are using `Put File` keyword here which is using SCP behind the scenes (non-root user), it cannot copy directly into the system folder with out escalating to root.